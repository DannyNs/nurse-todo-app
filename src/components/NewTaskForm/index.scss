.nr-new-task-form {

  div &__button,
  div &__button:hover {
    background-color: var(--main-success-color);

    span {
      color: var(--light-text-color);
    }
  }

  &__input {
    width: 100%;
  }
}