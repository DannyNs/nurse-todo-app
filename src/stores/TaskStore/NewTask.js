import {
  observable, action, computed,
} from 'mobx';

export default class NewTask {
  @observable
  name = '';

  @observable
  description = '';

  constructor(source) {
    Object.assign(this, source);
  }

  @action clearData = () => {
    this.name = '';
    this.description = '';
  }

  @action setName = (newName) => {
    this.name = newName;
  }

  @action setDescription = (newDescription) => {
    this.description = newDescription;
  }

  @computed get submitDisabled() {
    return !(this.name.length > 0 && this.description.length > 0);
  }
}
