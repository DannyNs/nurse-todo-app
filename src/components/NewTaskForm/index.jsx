import React from 'react';
import PropType from 'prop-types';
import { observer } from 'mobx-react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';

import NewTask from '../../stores/TaskStore/NewTask';

import './index.scss';

const NewTaskForm = observer(({
  addNewTask,
  newTask,
}) => (
  <div className="nr-new-task-form">
    <Grid container spacing={24}>

      <Grid item xs={5}>
        <Input
          placeholder="Name"
          name="name"
          value={newTask.name}
          className="nr-new-task-form__input"
          onChange={e => newTask.setName(e.target.value)}
        />
      </Grid>

      <Grid item xs={5}>
        <Input
          placeholder="Task"
          name="description"
          value={newTask.description}
          className="nr-new-task-form__input"
          onChange={e => newTask.setDescription(e.target.value)}
        />
      </Grid>

      <Grid item xs={2}>
        <Button variant="contained" className="nr-new-task-form__button" onClick={() => addNewTask(newTask)}>
          +
        </Button>
      </Grid>

    </Grid>
  </div>
));

NewTaskForm.propTypes = {
  addNewTask: PropType.func.isRequired,
  newTask: PropType.instanceOf(NewTask).isRequired,
};

export default NewTaskForm;
