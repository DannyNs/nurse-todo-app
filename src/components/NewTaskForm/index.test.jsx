import React from 'react';
import { shallow } from 'enzyme';
import NewTaskForm from '.';
import NewTask from '../../stores/TaskStore/NewTask';

describe('NewTaskForm', () => {
  it('should render without crashing', () => {
    // given
    const newTask = new NewTask();
    const addNewTask = () => {};

    // when
    const wrapper = shallow(
      <NewTaskForm addNewTask={addNewTask} newTask={newTask} />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
