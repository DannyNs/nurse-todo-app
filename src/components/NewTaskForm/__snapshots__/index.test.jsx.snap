// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskForm should render without crashing 1`] = `
<div
  className="nr-new-task-form"
>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={5}
    >
      <WithStyles(Input)
        className="nr-new-task-form__input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={5}
    >
      <WithStyles(Input)
        className="nr-new-task-form__input"
        name="description"
        onChange={[Function]}
        placeholder="Task"
        value=""
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={2}
    >
      <WithStyles(Button)
        className="nr-new-task-form__button"
        onClick={[Function]}
        variant="contained"
      >
        +
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
